### PR TITLE
Show correct metrics in row evolution triggered from goal view tables

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

This PR contains the following changes:

The row evolution will now show the goal metrics when a row evolution is opened from 
- a report on a goals report page (specific or overview) 
- the ecommerce sales page
- or any other report that has been switched to the goals visualization

Additionally it fixes this issues that existed before:

- When a row evolution was opened from a goals report page the url parameter contained the `idGoal`, loading this URL again resulted in an error page, as the previously used javascript did not remove the `idGoal` before sending the request. See eg https://demo.matomo.cloud/index.php?module=CoreHome&action=index&idSite=1&period=day&date=yesterday#?idSite=1&period=day&date=yesterday&segment=&category=Goals_Goals&subcategory=4&popover=RowAction$3ARowEvolution$3AReferrers.getReferrerType$3A$257B$2522idGoal$2522$253A$25224$2522$257D$3A$40Websites

- In a multi row evolution it is possible to choose the metric. Choosing the metric caused loosing any possible id parameters before, as the `extraParameters` were reset for this action. This caused a similar error as above, as an id param might be missing. This can be reproduced when e.g. switching the metric in a multi row evolution for forms: https://demo.matomo.cloud/index.php?module=CoreHome&action=index&idSite=1&period=day&date=yesterday#?idSite=1&period=day&date=yesterday&segment=&category=FormAnalytics_Forms&subcategory=7&popover=RowAction$3ARowEvolution$3AFormAnalytics.getPageUrls$3A$257B$2522action$2522$253A$2522getMultiRowEvolutionPopover$2522$252C$2522idForm$2522$253A$25227$2522$257D$3A$40divezone.net$252Fjobs$252Fapply$252Fchinese-speaking-ssi-instructor-or-padi-instructor$2C$40divezone.net$252Fjobs$252Fapply$252Fdivemaster-2

fixes #3883

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
This can be already reviewed if someone has some time. I'll try to add some additional UI tests tomorrow that cover the "bugs" I mentioned in the description to ensure they won't reoccur in the future.

File: plugins/API/RowEvolution.php [hunk 1]
```diff
@@ -90,6 +149,41 @@ public function getRowEvolution($idSite, $period, $date, $apiModule, $apiAction,
...
+                return $this->getAllGoalIds($idSite);
+            case Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_ORDER:
+            default:
+                return [$goalId];
+        }
+    }
+
+    protected function getAllGoalIds($idSite)
+    {
+        static $goalIds = [];
```
Comment (tsteur) [inline plugins/API/RowEvolution.php]: might need to use a transient cache here so it's automatically reset between tests?

**Review by tsteur — COMMENTED** [review]
Works 👍  Code looks good too, left only one comment

File: plugins/API/RowEvolution.php (hunk 1 above)
Comment (sgiehl) [reply plugins/API/RowEvolution.php]: Guess I will simply remove that caching again. Had used the method somewhere else before. But as it's not needed anymore and actually only called once in a request, there's no need to cache it.

Comment (sgiehl) [reply plugins/API/RowEvolution.php]: Ah no. It's the method that calls `getAllGoalIds`, which is still called twice. Will use Transient cache then

**Comment by sgiehl** [comment]
@tsteur updated the caching and also added some more tests for row evolution.
Also decided to move some of the goals related UI tests to the Goals plugin. Imho it's easier to have smaller units of tests than everything in the big `UIIntegration` class, as it's easier to execute those single tests locally then.

**Pull request merged**